packages/instantsearch.js/src/widgets/index/index.ts

```diff
@@ -584,6 +586,15 @@ const index = (widgetParams: IndexWidgetParams): IndexWidget => {
         return;
       }
 
+      // we can't attach a listener to the error event of search, as the error
+      // then would no longer be thrown for global handlers.
+      if (
+        instantSearchInstance.status === 'error' &&
+        !instantSearchInstance.mainHelper!.hasPendingRequests()
+      ) {
+        helper!.setState(lastValidSearchParameters!);
+      }
+
```

Comment:
I wonder if this can be put somewhere else? I thought in InstantSearch error handler, but that doesn't know local state etc, so would require a dedicated method on index widget.

I guess this could be problematic if render is called another time after error, with error again, it will again reset. (not sure if that behaviour is right?)